File: FlightService-BackEnd/FlightServiceAPI/Models/Ticket.cs
using FlightServiceAPI.Data;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FlightServiceAPI.Models
{
    public class Ticket
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ConfirmationNumber { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public string? SeatNumber { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public string? GateNumber { get; set; }
        public string? TicketClass { get; set; }
        public string? TicketPrice { get; set; }
        public int? PassengerId { get; set; }
        public int? FlightId { get; set; }
        public Flight? Flight { get; set; }
        public Passenger? Passenger { get; set; }
    }
}

File: FlightService-FrontEnd/models/ticket.ts
interface Ticket {
    confirmationNumber: number;
    seatNumber: string;
    gateNumber: string;
    ticketClass: string;
    ticketPrice: string;
    passengerId: string;
    flightId: string; 
    // passengerId: number;
    // flightId: number; 
}

export interface AddTicket {
    ticketClass: string;
    ticketPrice: string;
    passengerId: number;
    flightId: number; 
    //passengerId: number;
    //flightId: number; 
}

export default Ticket;
